### ISSUE TITLE
add ability to use plugins

### ISSUE THREAD

**Issue by soleiman** [Developer-Plexscape/ngx-grapesjs#23]
Hi,
Can i use plugins for ngx-grapes-webpage-editor ?

**Comment by bampakoa** [comment]
Hello @soleiman 👋 

In what type of plugins do you refer to?

**Comment by arimus** [comment]
You can use grapesjs plugins, which I'm guessing the question was. Here's an example of getting a reference to the component and passing the plugins along. It would be nice if there was a "plugins" property similar to the template / storagePrefix properties:

```
<ngx-grapes-webpage-editor #webpageEditorComponent [template]="template" />
```

```
import { AfterViewInit, Component, Inject, OnInit, PLATFORM_ID, ViewChild } from '@angular/core';
import { isPlatformBrowser, JsonPipe, Location, NgForOf, NgIf } from '@angular/common';
import { SafePipe } from '../../app/safe.pipe';
import { NgxWebpageEditorComponent } from 'ngx-grapesjs';

@Component({
  selector: 'app-builder',
  standalone: true,
  imports: [NgxWebpageEditorComponent],
  templateUrl: './builder.component.html',
  styleUrl: './builder.component.scss'
})
export class BuilderComponent implements OnInit, AfterViewInit {
  template = '<span>This is a test content</span>';
  @ViewChild('webpageEditorComponent') webpageEditorComponent: NgxWebpageEditorComponent | undefined;

  constructor(@Inject(PLATFORM_ID) private platformId: any) {}

  async ngOnInit() {
    if (isPlatformBrowser(this.platformId)) {
      // builder init here
    }
  }

  async ngAfterViewInit() {
    if (isPlatformBrowser(this.platformId)) {
      // builder init here
      this.webpageEditorComponent?.setup({
        container: "#gjs",
        plugins: [
          "grapesjs-preset-webpage", 
          "gjs-blocks-basic",
        ]
      });
    }
  }
}

```

**Comment by stevenrlp** [comment]
Is there anyway to use grapesjs-mjml plugin, if I try to add the plugin as a string i get an error of Plugin grapesjs-mjml not found.

**Comment by arimus** [comment]
> Is there anyway to use grapesjs-mjml plugin, if I try to add the plugin as a string i get an error of Plugin grapesjs-mjml not found.
> 
> 

Hrmm. And you have the 'grapesjs-mjml' plugin installed as a dep?

**Comment by stevenrlp** [comment]
> > Is there anyway to use grapesjs-mjml plugin, if I try to add the plugin as a string i get an error of Plugin grapesjs-mjml not found.
> 
> Hrmm. And you have the 'grapesjs-mjml' plugin installed as a dep?

I was able to get it working, I had to include the grapesjs-mjml script in the component body, but I found it caused a lot of console warnings so had to use an alternative solution.

**Comment by bampakoa** [comment]
@arimus that's a great suggestion! We will incorporate it soon in the next release.

@stevenrlp I would suggest to add the grapesjs-mjml script in the `scripts` property of the `angular.json` file.

**Comment by stevenrlp** [comment]
> [@stevenrlp](https://github.com/stevenrlp) I would suggest to add the grapesjs-mjml script in the `scripts` property of the `angular.json` file.

Yes that would work also, but wouldn't that add the script to every component instead of just the one component that needs it?

**Comment by bampakoa** [comment]
Yes, that's correct. If you want to load it only in one component, you should try use a dynamic import statement. Adding scripts in component template files is considered a bad practice in Angular.

**Comment by stevenrlp** [comment]
> Yes, that's correct. If you want to load it only in one component, you should try use a dynamic import statement. Adding scripts in component template files is considered a bad practice in Angular.

Yes that was what I was using, dynamic imports, thanks.
